installer/PowerToysBootstrapper/bootstrapper/bootstrapper.cpp

```diff
@@ -288,9 +296,31 @@ int WINAPI WinMain(HINSTANCE, HINSTANCE, LPSTR, int)
         sei.fMask = { SEE_MASK_FLAG_NO_UI | SEE_MASK_NOASYNC | SEE_MASK_NO_CONSOLE };
         sei.lpFile = newPTPath->c_str();
         sei.nShow = SW_SHOWNORMAL;
-        sei.lpParameters = UPDATE_REPORT_SUCCESS;
         ShellExecuteExW(&sei);
     }
 
     return 0;
 }
+
+int WINAPI WinMain(HINSTANCE, HINSTANCE, LPSTR, int)
+{
+    try
+    {
+        return bootstrapper();
+    }
+    catch (const std::exception& ex)
+    {
+        MessageBoxA(nullptr, ex.what(), "Unhandled stdexception encountered!", MB_OK | MB_ICONERROR);
+    }
+    catch (winrt::hresult_error const& ex)
+    {
+        winrt::hstring message = ex.message();
+        MessageBoxW(nullptr, message.c_str(), L"Unhandled winrt exception encountered!", MB_OK | MB_ICONERROR);
+    }
+    catch (...)
+    {
+        auto lastErrorMessage = get_last_error_message(GetLastError());
+        std::wstring message = lastErrorMessage ? std::move(*lastErrorMessage) : L"";
+        MessageBoxW(nullptr, message.c_str(), L"Unknown exception encountered!", MB_OK | MB_ICONERROR);
+    }
+}
```

Comment:
Warning	C4715	'WinMain': not all control paths return a value	bootstrapper	D:\git\andrey\PowerToys\installer\PowerToysBootstrapper\bootstrapper\bootstrapper.cpp	326